netebpfext/net_ebpf_ext_bind.c

```diff
@@ -74,7 +74,12 @@ static net_ebpf_extension_program_info_provider_t* _ebpf_bind_program_info_provi
 // Bind Hook NPI Provider.
 //
 ebpf_attach_provider_data_t _net_ebpf_bind_hook_provider_data = {
-    EBPF_ATTACH_PROVIDER_DATA_HEADER, EBPF_PROGRAM_TYPE_BIND_GUID, BPF_ATTACH_TYPE_BIND, BPF_LINK_TYPE_PLAIN};
+    .header = EBPF_ATTACH_PROVIDER_DATA_HEADER,
+    .supported_program_type = EBPF_PROGRAM_TYPE_BIND_GUID,
+    .bpf_attach_type = BPF_ATTACH_TYPE_BIND,
+    .link_type = BPF_LINK_TYPE_PLAIN,
+    .capabilities = {.support_extension_data_v1 = true},
```

Comment:
Would you expect to have a bit for every extension data version?  Or should there be a supported_version field?

Reply:
My thinking was that other errata besides supported versions number could be exposed here. Some extensions incorrectly fail to check the version on the ebpf_extension_data_t and as a result crash when presented with a later version. This flag was intended to detect if the extension can handle the new v1 structure without crashing.

For subsequent major version changes, the client and provider should instead first attempt the NMR negotiation with the latest version, if that fails then fall back to previous version. In this case that was not possible as the extension wasn't checking the version number correctly, hence the need for this flag check.